src/SFA.DAS.AssessorService.Application.Api/Controllers/CertificateQueryController.cs

```diff
@@ -40,6 +42,28 @@ public CertificateQueryController(IMediator mediator)
         public async Task<IActionResult> GetCertificates([FromQuery] List<string> statuses)
         {
             return Ok(await _mediator.Send(new GetCertificatesRequest { Statuses = statuses }));
-        }       
+        }
+
+        [HttpGet("addresses", Name = "GetPreviousAddresses")]
+        [SwaggerResponse((int)HttpStatusCode.OK, Type = typeof(List<CertificateAddress>))]
+        [SwaggerResponse((int)HttpStatusCode.InternalServerError, Type = typeof(ApiResponse))]
+        public async Task<IActionResult> GetPreviousAddresses([FromQuery] string userName)
+        {
+            var addresses = await _mediator.Send(new GetPreviousAddressesRequest { Username = userName });
+            if (addresses.Count == 0)
+                throw new ResourceNotFoundException();
```

Comment:
@jcoxhead Can this just return an empty array or something please? Or at the very least handle the 404 on the web app end if there are no previous addresses (which is a perfectly valid state).